src/renderer/components/cluster-manager/cluster-status.tsx

```diff
@@ -32,100 +32,92 @@ interface Dependencies {
 
 @observer
 class NonInjectedClusterStatus extends React.Component<ClusterStatusProps & Dependencies> {
-  @observable authOutput: KubeAuthUpdate[] = [];
-  @observable isReconnecting = false;
+  private readonly authOutputs = observable.map<ClusterId, KubeAuthUpdate[]>();
+  private readonly reconnecting = observable.set<ClusterId>();
 
-  constructor(props: ClusterStatusProps & Dependencies) {
-    super(props);
-    makeObservable(this);
-  }
-
-  get cluster(): Cluster {
-    return this.props.cluster;
-  }
+  private readonly authOutput = computed(() => this.authOutputs.get(this.clusterId) ?? []);
+  private readonly hasErrors = computed(() => this.authOutput.get().some(({ isError }) => isError));
+  private readonly isReconnecting = computed(() => this.reconnecting.has(this.clusterId));
+  private readonly clusterName = computed(() => {
+    const entity = catalogEntityRegistry.getById(this.clusterId);
+    const { cluster } = this.props;
 
-  @computed get entity() {
-    return catalogEntityRegistry.getById(this.cluster.id);
-  }
+    return entity?.getName() ?? cluster.name;
+  });
 
-  @computed get hasErrors(): boolean {
-    return this.authOutput.some(({ isError }) => isError);
+  private get clusterId() {
+    return this.props.cluster.id;
   }
 
   componentDidMount() {
     disposeOnUnmount(this, [
-      ipcRendererOn(`cluster:${this.cluster.id}:connection-update`, (evt, res: KubeAuthUpdate) => {
-        this.authOutput.push(res);
+      ipcRendererOn(`cluster:connection-update`, (evt, clusterId: ClusterId, update: KubeAuthUpdate) => {
+        getOrInsert(this.authOutputs, clusterId, []).push(update);
       }),
     ]);
   }
 
-  componentDidUpdate(prevProps: Readonly<ClusterStatusProps>): void {
-    if (prevProps.cluster.id !== this.props.cluster.id) {
-      this.isReconnecting = false;
-      this.authOutput = [];
-    }
-  }
-
-  reconnect = async () => {
-    this.authOutput = [];
-    this.isReconnecting = true;
+  private reconnect = async () => {
+    this.authOutputs.delete(this.clusterId);
+    this.reconnecting.add(this.clusterId);
 
     try {
-      await requestClusterActivation(this.cluster.id, true);
+      await requestClusterActivation(this.clusterId, true);
     } catch (error) {
-      this.authOutput.push({
+      getOrInsert(this.authOutputs, this.clusterId, []).push({
```

Comment:
Hmm. Using something called `getOrInsert` here is a bit convoluted, as nothing is never "got". Could here be a function called eg. `setAuthFailure` to document proper intent in this level of abstraction?

Reply:
More like `setAuthFailure(error);`, because `appendAuthUpdate` is just something technical without clear intent or motivation.